hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -173,9 +173,11 @@ public KafkaOffsetGen(TypedProperties props) {
     this.props = props;
 
     kafkaParams = new HashMap<>();
-    for (Object prop : props.keySet()) {
+    props.keySet().stream().filter(prop -> {
```

Comment:
@li36909 This change does not change any logging unlike what the title of the PR says, can you correct the title and explain this change please ?

Reply:
how about change to this: "DeltaStream print many unnecessary warn log because of passing hoodie config to kafka consumer". the warn log is printed by kafkaconsumer. when hudi new the kafka consumer, hudi pass some non-kafka parameter to the kafka comsumer, then lead to these warn log, to solve this problem we just need to filter the hoodie config.

Reply:
BTW, I find a UT fail cause by concurrent write to a hudi table, I will try to analyze it later

Reply:
@li36909 Okay, I see. The current title makes sense to me now. Can you leave the above comment you made in the code as to why we are "removing the hoodie properties" ? Once that is done and the CI is green, we can land this.

Reply:
@n3nash I had added comment at the code, thank you